src/components/home/Trend/Trend.tsx

```diff
@@ -0,0 +1,169 @@
+import { productListMock } from '@/data/productListMock'
+import { theme } from '@/styles/theme'
+import styled from '@emotion/styled'
+import { useState, useEffect } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import type { TargetType, RankType } from './types'
+import { isValidTargetType, isValidRankType } from './types'
+import { TrendFilter } from './TrendFilter'
+import { ProductItem } from './ProductItem'
+
+// * 실시간 급상승 컨테이너
+const Container = styled.div`
+  width: 100%;
+  height: fit-content;
+
+  padding: 20px;
+
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: flex-start;
+  gap: 20px;
+`
+
+// * 실시간 급상승 상품 컨테이너
+const ProductContainer = styled.div`
+  width: 100%;
+  height: fit-content;
+
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  column-gap: ${theme.spacing.spacing2};
+  row-gap: ${theme.spacing.spacing5};
+`
+
+// * 더보기 버튼 컨테이너
+const MoreButtonContainer = styled.div`
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin-top: ${theme.spacing.spacing4};
+`
+
+// * 더보기 버튼
+const MoreButton = styled.button`
+  max-width: 480px;
+  width: 100%;
+  height: fit-content;
+
+  padding: ${theme.spacing.spacing3} ${theme.spacing.spacing6};
+  background-color: ${theme.colors.gray.gray00};
+  border: 1px solid ${theme.colors.gray.gray400};
+  border-radius: ${theme.spacing.spacing1};
+
+  font-size: ${theme.typography.body.body2Regular.fontSize};
+  font-weight: ${theme.typography.body.body2Regular.fontWeight};
+  line-height: ${theme.typography.body.body2Regular.lineHeight};
+  color: ${theme.colors.gray.gray800};
+
+  cursor: pointer;
+  transition: all 0.2s ease;
+
+  &:hover {
+    background-color: ${theme.colors.gray.gray100};
+    border-color: ${theme.colors.gray.gray500};
+  }
+
+  &:active {
+    background-color: ${theme.colors.gray.gray200};
+  }
+`
+
+// * 실시간 급상승 컴포넌트
+export const Trend = () => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [showAll, setShowAll] = useState(false)
+
+  // URL 파라미터에서 초기값 가져오기
+  const getInitialTargetType = (): TargetType => {
+    const urlTargetType = searchParams.get('targetType')
+    return urlTargetType && isValidTargetType(urlTargetType) ? urlTargetType : 'ALL'
+  }
+
+  const getInitialRankType = (): RankType => {
+    const urlRankType = searchParams.get('rankType')
+    return urlRankType && isValidRankType(urlRankType) ? urlRankType : 'MANY_WISH'
+  }
+
+  const [targetType, setTargetType] = useState<TargetType>(getInitialTargetType)
+  const [rankType, setRankType] = useState<RankType>(getInitialRankType)
+
+  const INITIAL_SHOW_COUNT = 6
+
+  // URL 파라미터 변경 감지
+  useEffect(() => {
+    const urlTargetType = searchParams.get('targetType')
+    const urlRankType = searchParams.get('rankType')
+
+    if (urlTargetType && isValidTargetType(urlTargetType) && urlTargetType !== targetType) {
```

Comment:
```suggestion
    if (urlTargetType && isValidTargetType(urlTargetType)) {
```

` urlTargetType !== targetType`
사실 이 비교는 불필요합니다.
setState에서 같은 값이면 어차피 렌더링을 일으키지 않고, targetType 값은 string 타입, 즉 primitive 타입이기 때문에 값을 비교하는 것도 안전하고요.

Reply:
뭔가 더 타입을 상세히 검증해보고 싶었는데, 불필요한 코드는 없애는게 좋을 것 같다는 것을 다시 느끼게 되었습니다. 조언 감사합니다!